private-templates-service/client/src/components/NavBar/NavBar.tsx

```diff
@@ -96,14 +115,15 @@ const NavBar = (props: NavBarProps) => {
     default:
       return (
         <Banner>
-          <MobileBanner>
+          <BaselineBanner>
             <StyledLogo src={Logo} />
-            <Styledh1>{props.currentPageTitle}</Styledh1>
-          </MobileBanner>
+            <Styledh1>{props.templateName || props.currentPageTitle}</Styledh1>
+            {!props.isFetching && <EditButton onClick={editName} iconProps={{ iconName: 'Edit' }} />}
```

Comment:
here, are you saying that the default case will be used for the template preview page? if so, does this imply that name is only editable from that page? do we not want to allow name change from designer page?

Reply:
khang is working on that part

Reply:
okay sure. but i still feel like we should add a new case in this switch statement to handle the PreviewModal page instead of using the default case.

Reply:
agreed, done